helix-core/src/test/java/org/apache/helix/integration/rebalancer/WagedRebalancer/TestWagedRebalance.java

```diff
@@ -354,12 +354,12 @@ public void testMaxPartitionLimitation() throws Exception {
       String limitedResourceName = null;
       int i = 0;
       for (String stateModel : _testModels) {
-        String db = "Test-DB-" + i++;
+        String db = "Test-DB-testMaxPartitionLimitation-" + i++;
         createResourceWithWagedRebalance(CLUSTER_NAME, db, stateModel, PARTITIONS, _replica,
             _replica);
         if (i == 1) {
           // The limited resource has additional limitation, so even the other resources can be assigned
-          // later, this resource will still be blocked by the max partition limitation.
+          // later in theory, this resource will still be blocked by the max partition limitation.
```

Comment:
I am a little confused. Is the additional limitation maxPartitionsPerInstance(1) you're referring to?

Reply:
Let me refine the comments.
This is special for the WAGED rebalancer.